mongo/bulk_write.go

```diff
@@ -264,7 +264,7 @@ func createDeleteDoc(filter interface{}, collation *options.Collation, hint inte
 		doc = bsoncore.AppendDocumentElement(doc, "collation", collation.ToDocument())
 	}
 	if hint != nil {
-		hintVal, err := transformValue(registry, hint)
+		hintVal, err := transformValue(registry, hint, false)
```

Comment:
As long as the final error message is informative as to what function passed in an invalid map, it seems fine to me if the error is generated in `transformValue()`. You could append the function name in the error check directly below? Something like
```
return nil, fmt.Errorf("error transforming hint value: %v", err)
```
That goes for whenever `mapAllowed` is false.

Reply:
We've already sort of set a precedent with the ErrNilValue on mongo.go/440, that this might break. For example currently, if a nil value is passed in to SetHint or SetSort, the only information you get back is that ErrNilValue, and if we change those to report the erroring field we should change all the error code for transforming a value to report the field. I agree that it would be useful, but it would be a bigger change than this ticket implies.

Reply:
If it's a larger change I am ok leaving it as is. I am just happy we are protecting users from incorrect results!